put method generates "Failed SFTP MKDIR"

Initially reported on gradle-ssh-plugin. See https://github.com/int128/gradle-ssh-plugin/issues/285

Version: 2.7.2

Operations:
```
ssh.run {
  session(remotes.syf) {
     put from: "${project.rootDir}/builder/${project.name}", into: '/home/syf/sshupload'
  }
}
```

Console:
```
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/grp
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/grp/grwb
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/grp/gspi
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/grp/grwb
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/grp/gspi
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm/cpws
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm/tdtf
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm/tspi
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm/cpws
Failed SFTP MKDIR: syf:/home/syf/sshupload/tax_settle/jsp/lcm/tdtf

```
but files are uploaded.

With --debug, we can see the files are uploaded several times leading to several calls to mkdir:
```11:15:20.058 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Requesting SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250
11:15:20.058 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Success SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250
11:15:20.058 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Requesting SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:20.058 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Success SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:20.058 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Requesting SFTP PUT: D:\temp\data\20170201-111504\bestofPV_inc\output_1474602250\4875\index.html -> filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:20.074 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Success SFTP PUT: D:\temp\data\20170201-111504\bestofPV_inc\output_1474602250\4875\index.html -> filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:20.074 [INFO] [org.hidetake.groovy.ssh.operation.SftpOperations] Sent file to filer: D:\temp\data\20170201-111504\bestofPV_inc\output_1474602250\4875\index.html -> /tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
[...]
11:15:21.949 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Requesting SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:21.964 [INFO] [org.hidetake.groovy.ssh.operation.SftpOperations] Remote directory already exists on filer: /tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:21.964 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Requesting SFTP PUT: D:\temp\data\20170201-111504\bestofPV_inc\output_1474602250\4875\index.html -> filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:21.964 [ERROR] [org.hidetake.groovy.ssh.operation.SftpOperations] Failed SFTP MKDIR: filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
11:15:21.980 [DEBUG] [org.hidetake.groovy.ssh.operation.SftpOperations] Success SFTP PUT: D:\temp\data\20170201-111504\bestofPV_inc\output_1474602250\4875\index.html -> filer:/tmp/data/tmp/20170201-111504/bestofPV_inc/output_1474602250/4875
```
This bug increases a lot the time transfer.

--- a/core/src/main/groovy/org/hidetake/groovy/ssh/session/transfer/put/Instructions.groovy
+++ b/core/src/main/groovy/org/hidetake/groovy/ssh/session/transfer/put/Instructions.groovy
@@ -1,5 +1,7 @@
 package org.hidetake.groovy.ssh.session.transfer.put
 
+import groovy.io.FileType
+
 /**
  * Represents a collection of SCP PUT instructions
  * such as {@link File}, {@link StreamContent}, {@link EnterDirectory} or {@link LeaveDirectory}.
@@ -50,7 +52,7 @@ class Instructions implements Iterable {
         if (path.directory) {
             def children = []
             children.add(new EnterDirectory(path.name))
-            path.eachFile { file -> children.addAll(forFileRecursive(file)) }
+            path.eachFile(FileType.FILES) { file -> children.addAll(forFileRecursive(file)) }
             path.eachDir { dir -> children.addAll(forFileRecursive(dir)) }
             children.add(LeaveDirectory.instance)
             children
@@ -74,7 +76,7 @@ class Instructions implements Iterable {
     private static Collection forFileWithFilterRecursive(File path, Closure<Boolean> filter) {
         if (path.directory) {
             def children = new ArrayDeque()
-            path.eachFile { file -> children.addAll(forFileWithFilterRecursive(file, filter)) }
+            path.eachFile(FileType.FILES) { file -> children.addAll(forFileWithFilterRecursive(file, filter)) }
             path.eachDir { dir -> children.addAll(forFileWithFilterRecursive(dir, filter)) }
             if (!children.empty) {
                 children.addFirst(new EnterDirectory(path.name))

